cmd/ps3netsrv-go/handler.go

```diff
@@ -22,18 +24,70 @@ func (h *Handler) HandleOpenDir(ctx *server.Context, path string) bool {
 	log := h.Log.With(zap.Stringer("remote", ctx.RemoteAddr), zap.String("path", path))
 	log.Info("Open dir")
 
-	info, err := h.Fs.Stat(path)
+	handle, err := h.Fs.Open(path)
+	if err != nil {
+		log.Warn("Open failed", zap.Error(err))
+		return false
+	}
+
+	info, err := handle.Stat()
 	if err != nil {
 		log.Warn("Stat failed", zap.Error(err))
 		return false
 	}
 
+	if ctx.CwdHandle != nil {
+		if err := ctx.CwdHandle.Close(); err != nil {
```

Comment:
I think it will be better to set `CwdHandle` to nil after close (like it's implemented in `Context.Close`) in connection close procedure) here and further to avoid possible double closing

Reply:
Fixed